recipes/libpciaccess/all/conanfile.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+
+class LibPciAccessConan(ConanFile):
+    name = "libpciaccess"
+    description = "Generic PCI access library"
+    topics = ("pci", "xorg")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gitlab.freedesktop.org/xorg/lib/libpciaccess"
+    license = "MIT", "X11"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _autotools = None
+
+    def validate(self):
+        def is_supported(settings):
+            if settings.os in ("Linux", "FreeBSD", "SunOS"):
+                return True
+            return settings.os == "Windows" and settings.get_safe("os.subsystem") == "cygwin"
+        if not is_supported(self.settings):
+            raise ConanInvalidConfiguration("Unsupported architecture.")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        pass
+
+    def build_requirements(self):
+        self.build_requires("pkgconf/1.7.4")
+        self.build_requires("xorg-macros/1.19.3")
+        self.build_requires("libtool/2.4.6")
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.default_configure_install_args = True
+        tc.generate()
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = Autotools(self)
+        self._autotools.configure(build_script_folder=self._source_subfolder)
+        return self._autotools
```

Comment:
Do we still need this anti-pattern when using `Autotools` instead of `AutotoolsBuildEnvironment`?
Can we do the following yet?
```python
def build(self):
    autotools = Autotools(self)
    autotools.configure()
    autotools.make()
def package(self):
    autotools = Autotools(self)
    autotools.install()
```
I mean, running `conan create . libpciaccess/0.16@ -kb` should not run the configure script again.
I remember reading in a recent conan blog about the above becoming possible.

Reply:
you're referring to https://blog.conan.io/2021/10/19/New-CMake-Transparent-Integration-Guide.html

Reply:
That's the one. Thanks!